### PR TITLE
Fix warnings due to improper use of strncpy()

### PR DESCRIPTION
**Pull request by keithc-ca**
Use of `strncpy(dst, src, strlen(src))` yields a warning.

### CODE REVIEW

File: runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp [hunk 1]
```diff
@@ -308,28 +309,32 @@ getJ9UTF8SignatureFromMethodType(J9VMThread *currentThread, j9object_t typeObjec
...
 	cursor = (char*)J9UTF8_DATA(methodDescriptor);
+	space = signatureLength;
 	*cursor++ = '(';
+	space -= 1;
 
 	/* Copy class signatures to descriptor string */
 	for (U_32 i = 0; i < numArgs; i++) {
 		UDATA len = strlen(signatures[i]);
-		strncpy(cursor, signatures[i], len);
+		strncpy(cursor, signatures[i], space);
```
Comment (pshipton) [inline runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp:329]: It's not very efficient, as it's padding with nulls to `space` every time through the loop.

Comment (keithc-ca) [reply runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp:329]: It's always seemed unnatural to me that strncpy has that behavior, but you're right, I'll fix that.

**Comment by genie-openj9** [comment]
Can one of the admins verify this patch?

**Comment by keithc-ca** [comment]
Updated, also addresses:
* fix signature and allow for failure of `getJ9FormattedName()`
* fix warnings initializing `TR_Version`
* clean up `testjep178`

File: runtime/compiler/runtime/RelocationRuntime.cpp [hunk 2]
```diff
@@ -1064,12 +1064,13 @@ TR_SharedCacheRelocationRuntime::validateAOTHeader(TR_FrontEnd *fe, J9VMThread *
       intptr_t featureFlags = generateFeatureFlags(fe);
 
       TR_Version currentVersion;
-      memset(&currentVersion, 0, sizeof(TR_Version));
       currentVersion.structSize = sizeof(TR_Version);
       currentVersion.majorVersion = TR_AOTHeaderMajorVersion;
       currentVersion.minorVersion = TR_AOTHeaderMinorVersion;
-      strncpy(currentVersion.vmBuildVersion, EsBuildVersionString, std::min(strlen(EsBuildVersionString), sizeof(currentVersion.vmBuildVersion) - 1));
-      strncpy(currentVersion.jitBuildVersion, TR_BUILD_NAME, std::min(strlen(TR_BUILD_NAME), sizeof(currentVersion.jitBuildVersion) - 1));
+      strncpy(currentVersion.vmBuildVersion, EsBuildVersionString, sizeof(currentVersion.vmBuildVersion) - 1);
```
Comment (pshipton) [inline runtime/compiler/runtime/RelocationRuntime.cpp:1070]: This is doing unnecessary padding now.

File: runtime/compiler/runtime/RelocationRuntime.cpp [hunk 3]
```diff
@@ -1064,12 +1064,13 @@ TR_SharedCacheRelocationRuntime::validateAOTHeader(TR_FrontEnd *fe, J9VMThread *
...
+      currentVersion.vmBuildVersion[sizeof(currentVersion.vmBuildVersion) - 1] = '\0';
+      strncpy(currentVersion.jitBuildVersion, TR_BUILD_NAME, sizeof(currentVersion.jitBuildVersion) - 1);
```
Comment (pshipton) [inline runtime/compiler/runtime/RelocationRuntime.cpp:1072]: Here as well.

File: runtime/compiler/runtime/RelocationRuntime.cpp [hunk 4]
```diff
@@ -1144,13 +1145,14 @@ TR_SharedCacheRelocationRuntime::createAOTHeader(TR_FrontEnd *fe)
       aotHeaderVersion->structSize = sizeof(TR_Version);
       aotHeaderVersion->majorVersion = TR_AOTHeaderMajorVersion;
       aotHeaderVersion->minorVersion = TR_AOTHeaderMinorVersion;
-      strncpy(aotHeaderVersion->vmBuildVersion, EsBuildVersionString, sizeof(EsBuildVersionString));
-      strncpy(aotHeaderVersion->jitBuildVersion, TR_BUILD_NAME, std::min(strlen(TR_BUILD_NAME), sizeof(aotHeaderVersion->jitBuildVersion)));
+      strncpy(aotHeaderVersion->vmBuildVersion, EsBuildVersionString, sizeof(aotHeaderVersion->vmBuildVersion) - 1);
```
Comment (pshipton) [inline runtime/compiler/runtime/RelocationRuntime.cpp:1148]: More padding. and below as well.

**Comment by pshipton** [comment]
I think the commit comment should be updated to better reflect the changes.

File: runtime/compiler/runtime/RelocationRuntime.cpp (hunk 2 above)
Comment (keithc-ca) [reply runtime/compiler/runtime/RelocationRuntime.cpp:1070]: The padding done by these uses of `strncpy()` is equivalent to the result via the removed call to `memset()`.

Comment (pshipton) [reply runtime/compiler/runtime/RelocationRuntime.cpp:1070]: Ok, but it could be done better.

**Comment by pshipton** [comment]
@fjeremic @dsouzai this will need JIT review(ers)

File: runtime/compiler/control/J9Options.cpp [hunk 5]
```diff
@@ -2133,19 +2133,21 @@ static TR::FILE *fileOpen(TR::Options *options, J9JITConfig *jitConfig, char *na
...
-     }
-  }
-
+      {
+      formattedTmp = TR_J9VMBase::getJ9FormattedName(jitConfig, PORTLIB, tmp, sizeof(tmp), name, options->getSuffixLogsFormat(), true);
+      }
+   if (NULL != formattedTmp)
+   {
+      return j9jit_fopen(formattedTmp, permission, b1, b2);
+   }
```
Comment (dsouzai) [inline runtime/compiler/control/J9Options.cpp:2148]: Formatting.

**Review by dsouzai — COMMENTED** [review]
LGTM excpet for a minor formatting inconsistency.

File: runtime/compiler/control/J9Options.cpp (hunk 5 above)
Comment (keithc-ca) [reply runtime/compiler/control/J9Options.cpp:2148]: Right - I'll fix that.

File: runtime/compiler/runtime/RelocationRuntime.cpp (hunk 2 above)
Comment (pshipton) [reply runtime/compiler/runtime/RelocationRuntime.cpp:1070]: To clarify, maybe the buffer doesn't need to be entirely cleared, but I'll leave that up to JIT review.

**Comment by keithc-ca** [comment]
Commit message updated.

**Comment by pshipton** [comment]
jenkins test sanity all jdk8,jdk11

**Review by pshipton — APPROVED** [review]

**Review by fjeremic — APPROVED** [review]

**Pull request merged**